pkgs/by-name/zx/zxpy/package.nix

```diff
@@ -2,35 +2,25 @@
   lib,
   python3,
   fetchFromGitHub,
-  fetchpatch,
   deterministic-uname,
 }:
-
-python3.pkgs.buildPythonApplication rec {
+let
+  version = "1.6.4";
+in
```

Comment:
I always find this kind of "let in" construct to be ugly, there's nothing wrong with rec as long as it's not overused.

Reply:
I see, I understand your point. However, I think that reevaluation occurs and leads to confusion by `rec` is true to some extent. How an above expression is often seen in the programming languages include `let ~ in`, for example OCaml.
But I don't have enough experience with nixpkgs, so could you let me know if there's anything I'm misunderstanding?

Reply:
What do you specifically mean by "reevaluation"?

```nix
rec {
  a = 1;
  b = 2;
  c = a + b;
}
```
is functionally identical to
```nix
let
  a = 1;
  b = 2;
  c = a + b;
in
{
  inherit a b c;
}
```
Yes, it does bring more into the "recursive" context than
```nix
let
  a = 1;
  b = 2;
in
{
  inherit a b;
  c = a + b;
}
```
Thus `rec` is objectively worth than a proper `let in`, but the aesthetic of such a construct is just, a sore to the eyes.

A middle ground is the `finalAttrs` pattern, where you have to explicitly use `finalAttrs.<attr>` to reference recursed attributes. But it's not supported by `buildPythonApplication` yet....

Reply:
Thank you for teaching your perspective on `let ... in` versus `rec`. 
I see your point that `let ... in` might be less aesthetically pleasing and that `rec` isn’t bad if used judiciously. 
I’ve updated the code to use `rec` instead, and it still works the same way. 
Could you please review the revised version?